src/main/java/duke/components/Parser.java

```diff
@@ -0,0 +1,227 @@
+            return ui.showByeMessage();
+
+        case LIST:
+            if (userInputParts.length > 1) {
+                return ui.showInvalidMessage();
+            }
+            return ui.showList(fullList);
+
+        case MARK:
+            String[] splitIndexMark = input.split(" ");
+            if (splitIndexMark.length <= 1 || splitIndexMark.length > 2) {
+                return ui.showInvalidMessage();
+            }
+            int index = Integer.parseInt(splitIndexMark[1]) - 1;
+            return fullList.markItem(index);
+
+        case UNMARK:
+            String[] splitIndexUnMark = input.split(" ");
+            if (splitIndexUnMark.length <= 1 || splitIndexUnMark.length > 2) {
+                return ui.showInvalidMessage();
+            }
+            int indexUnmark = Integer.parseInt(splitIndexUnMark[1]) - 1;
+            return fullList.unMarkItem(indexUnmark);
+
+        case TODO:
+            if (userInputParts.length <= 1) {
+                return ui.showInvalidMessage();
+            }
+            ToDos toDo = new ToDos(userInputParts[1].trim());
+            return fullList.addToList(toDo);
+
+        case DEADLINE:
+            String[] details = userInputParts[1].split("/by");
+            String description = details[0].trim();
+
+            if (details.length <= 1) {
+                return ui.showInvalidMessage();
+            }
+
+            LocalDateTime by = setDate(details[1].trim());
+            Deadline deadline = new Deadline(description, by);
+            return fullList.addToList(deadline);
+
+        case EVENT:
+            String[] det = userInputParts[1].split("/from");
+            if (det.length <= 1) {
+                return ui.showInvalidMessage();
+            }
+            String[] dateParts = det[1].trim().split("/to");
+            if (dateParts.length <= 1) {
+                return ui.showInvalidMessage();
+            }
+
+            String descr = det[0].trim();
+            LocalDateTime from = setDate(dateParts[0].trim());
+            LocalDateTime to = setDate(dateParts[1].trim());
+
+            Event event = new Event(descr, from, to);
+
+            if (event.isStartDateBefore(from, to)) {
+                return fullList.addToList(event);
+            } else {
+                return ui.showInvalidDate();
+            }
+
+        case DELETE:
+            String[] split = input.split(" ");
+            if (split.length <= 1 || split.length > 2) {
+                return ui.showInvalidMessage();
+            }
+            int ind = Integer.parseInt(split[1]) - 1;
+            return fullList.deleteFromList(ind);
+
+        case FIND:
+            String[] inputParts = input.split(" ", 2);
+            String keyword = inputParts[1].trim();
+            TaskList list = fullList.findTask(keyword);
+            if (list.getSize() == 0) {
+                return ui.showNoFind();
+            } else {
+                return ui.showFind(list);
+            }
+
```

Comment:
I like the use of line breaks to separate logical blocks and make the code easier to read!